src/OrleansRuntime/GrainTypeManager/GrainTypeData.cs

```diff
@@ -127,5 +128,19 @@ internal static MultiClusterRegistrationStrategy GetMultiClusterRegistrationStra
                             grainClass.Name));
             }
         }
+
+        // used by Orleankka's bootstrapper
+        internal void SetMessageAlwaysInterleaves(Func<object, bool> predicate)
```

Comment:
> grain implementation or interface

implementation. I think interface is not a good candidate since it's reentrancy is an implementation detail.